packages/toolpad-components/src/components/ErrorOverlay.tsx

```diff
@@ -0,0 +1,30 @@
+import { Typography, styled } from '@mui/material';
+import * as React from 'react';
+import { errorFrom } from '@mui/toolpad-utils/errors';
+
+const OverlayRoot = styled('div')(({ theme }) => ({
+  position: 'absolute',
+  inset: '0 0 0 0',
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  justifyContent: 'center',
+  borderWidth: 1,
+  borderStyle: 'solid',
+  borderRadius: theme.shape.borderRadius,
+  borderColor: theme.palette.divider,
+}));
+
+interface ErrorOverlayProps {
+  error?: unknown;
+}
+
+export default function ErrorOverlay({ error }: ErrorOverlayProps) {
+  const errMEssage = error ? errorFrom(error).message : null;
+  return (
+    <OverlayRoot sx={{ display: error ? undefined : 'none' }}>
+      <Typography>Error</Typography>
+      <Typography variant="body2">{errMEssage}</Typography>
```

Comment:
What if there's no error message? Should we have a fallback generic error message like `There was an unexpected error`?

Reply:
The `errorFrom` function ensures a valid error

Reply:
Ah, from a quick look at the logic I thought `errMessage` could be `null`, but I guess that only happens if there's no error... So it's fine then.
At first thought I would make the `error` prop be required for this component and only render the component if there is an error to pass as a prop. But if you're using the approach with the CSS `display` for a different reason it could be fine too.